homeassistant/components/systemmonitor/sensor.py

```diff
@@ -300,66 +455,33 @@ class SystemMonitorSensor(SensorEntity):
     def __init__(
         self,
         sensor_registry: dict[tuple[str, str], SensorData],
-        sensor_type: str,
+        sensor_description: SystemMonitorSensorEntityDescription,
         argument: str = "",
     ) -> None:
         """Initialize the sensor."""
-        self._type: str = sensor_type
-        self._name: str = f"{self.sensor_type[SENSOR_TYPE_NAME]} {argument}".rstrip()
-        self._unique_id: str = slugify(f"{sensor_type}_{argument}")
+        self.entity_description = sensor_description
+        self._attr_name: str = f"{sensor_description.name} {argument}".rstrip()
+        self._attr_should_poll = False
+        self._attr_unique_id: str = slugify(f"{sensor_description.key}_{argument}")
         self._sensor_registry = sensor_registry
+        self._type: str = sensor_description.key
```

Comment:
Replace the use of `self._type` with `self.entity_description.key`, allowing to remove this variable.